### PR TITLE
Docs: DRY up CRUD docs

### PR DESCRIPTION
**Pull request by nik9000**
This further applies the pattern set in #34125 to reduce copy-and-paste
in the single document CRUD portion of the High Level REST Client docs.
It also adds line wraps to snippets that are too wide to fit into the box
when rendered in the docs, following up on the work started in #34163.

### CODE REVIEW

File: docs/java-rest/high-level/execution.asciidoc [hunk 1]
```diff
@@ -38,7 +38,7 @@ completed the `ActionListener` is called back using the `onResponse` method
 if the execution successfully completed or using the `onFailure` method if
 it failed.
 
-A typical listener for +{response}+ looks like:
+A typical listener for +{api}+ looks like:
```
Comment (nik9000) [inline docs/java-rest/high-level/execution.asciidoc]: This wording sounds better for the exists API.

File: docs/java-rest/high-level/supported-apis.asciidoc [hunk 2]
```diff
@@ -11,6 +11,7 @@ The Java High Level REST Client supports the following Document APIs:
 Single document APIs::
 * <<{upid}-index>>
 * <<{upid}-get>>
+* <<{upid}-exists>>
```
Comment (nik9000) [inline docs/java-rest/high-level/supported-apis.asciidoc]: This was missing.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/CRUDDocumentationIT.java [hunk 3]
```diff
@@ -548,12 +557,13 @@ public void testDelete() throws Exception {
             // tag::delete-request
             DeleteRequest request = new DeleteRequest(
                     "posts",    // <1>
-                    "doc",     // <2>
-                    "1");      // <3>
+                    "doc",      // <2>
```
Comment (nik9000) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/CRUDDocumentationIT.java]: These didn't line up and they made me uncomfortable.

**Review by hub-cap — APPROVED** [review]
<3 ^ infinity

**Pull request merged**